target/linux/bmips/dts/bcm63168-sagem-fast-3864-op.dts

```diff
@@ -0,0 +1,310 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "bcm63268.dtsi"
+
+/ {
+	model = "Sagemcom F@ST 3864 OP";
+	compatible = "sagem,fast-3864-op", "brcm,bcm63168", "brcm,bcm63268";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_red;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_green;
+
+		led-dsl = &led_dsl_green;
+		led-internet = &led_internet_green;
+		led-usb = &led_usb_green;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 32 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+
+		wlan {
+			label = "wlan";
+			gpios = <&gpio 34 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WLAN>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ethernet {
+	status = "okay";
+
+	nvmem-cells = <&macaddr_cferom_6a0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&leds {
+	status = "okay";
+	brcm,serial-leds;
+	brcm,serial-dat-low;
+	brcm,serial-shift-inv;
+	brcm,serial-mux;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_leds &pinctrl_serial_led>;
+
+	led@0 {
+		reg = <0>;
+		active-low;
+		function = LED_FUNCTION_WPS;
+		color = <LED_COLOR_ID_GREEN>;
+	};
+
+	led@1 {
+		reg = <1>;
+		active-low;
+		function = LED_FUNCTION_WPS;
+		color = <LED_COLOR_ID_RED>;
+	};
+
+	led@2 {
+		reg = <2>;
+		active-low;
+		label = "red:internet";
+	};
+
+	led_dsl_green: led@3 {
+		reg = <3>;
+		active-low;
+		label = "green:dsl";
+	};
+
+	led@4 {
+		reg = <4>;
+		active-low;
+		label = "green:fxs";
+	};
+
+	led@5 {
+		reg = <5>;
+		active-low;
+		label = "red:fxs";
+	};
+
+	led@6 {
+		reg = <6>;
+		active-low;
+		label = "green:dummy6";
+	};
+
+	led@7 {
+		reg = <7>;
+		active-low;
+		label = "green:dummy7";
+	};
+
+	led_internet_green: led@8 {
+		reg = <8>;
+		active-low;
+		label = "green:internet";
+	};
+
+	led_usb_green: led@9 {
+		reg = <9>;
+		active-low;
+		label = "green:dsl_bonding";
```

Comment:
What's the difference between this and `led_dsl_green: led@3`?
Shouldn't this be `label = "green:usb";`?

Reply:
They are different LEDs. In the stock firmware, they were the indicators of the DSL modem status. On the PCB, they were labeled as "dsl" and "dsl_bonding". I prefer to keep their name so that whoever looks at the names won't get lost.

There is no dedicated LED to indicate USB activities and I repurposed LED "green:dsl_bonding" as the USB activity indicator, as it is the one closest to the USB port.

Reply:
> They are different LEDs. In the stock firmware, they were the indicators of the DSL modem status. On the PCB, they were labeled as "dsl" and "dsl_bonding". I prefer to keep their name so that whoever looks at the names won't get lost.

Yeah, that's the correct way to go in that case.

> There is no dedicated LED to indicate USB activities and I repurposed LED "green:dsl_bonding" as the USB activity indicator, as it is the one closest to the USB port.

In this case then the alias should still be `led_dsl_bonding_green` instead of `led_usb_green`.
However, since this can be configured manually by the user and it's not the original LED purpose, it's better to leave it up to the user to decide which LED to use for USB.